app/controllers/users_controller.rb

```diff
@@ -345,6 +345,16 @@ def shortlink
     end
   end
 
+  def verify_email
+    @user = User.find(params[:id])
```

Comment:
Can you refactor your code a bit so that we first decrypt the data received and on the successful decryption, check the user id and time ?

Reply:
But what would you compare your user id with?We have to make sure that the link generated for that user verifies his email. Would you mind providing a small example, that would make things clear. And if you don't want the user to to know his user_id , then we can use his username instead. Then I guess everything would work fine?

Reply:
We won't really need user_id and username.

Let me give you an example:

URL would look like `publiclab.org/profile/verify/here_comes_token`

We will get that token and will call `validate_token` class method which will return user in case token is valid, else false in case token is invalid. 

How do you define validity of token ? We just check whether timestamp is not too old(24hrs) and data decryption is successful. 

Then from the decrypted data we get the user_id and then return the user object.

Reply:
Okay so I have understood what you want to say, but in this case we ll have to make encrypt and decrypt methods that you implemented also class methods and not instance methods. So I can do that, not a problem. I am beginner with rails or in general with web development so I have this doubt: What is the problem in using user_id or username in the endpoint?

Reply:
> encrypt and decrypt methods that you implemented also class methods and not instance methods

I think this won't be required.

> What is the problem in using user_id or username in the endpoint?

Good question! Actually it doesn't affect anything. It's totally fine to use them in the endpoints but since we don't need them, it's redundant. Though about using user_id, if you check any other website like facebook, they never show original user id in their endpoints. You will see string of random chars which represent an item uniquely on their system, but why ? May be they don't want anyone to know the original database id. It's just considered as bad practise to show internal db ids.